### PR TITLE
generate: add option to specify some constraints

### PR DESCRIPTION
**Pull request by Melyodas**
This allows to personalize output more and choose more simple or
complicated input
only works with integer as it is not necessary for chars.

### CODE REVIEW

File: iorgen/random_input.py [hunk 1]
```diff
@@ -40,14 +41,28 @@ def eval_var(self, var: Union[int, str, Variable]) -> int:
...
             value = random.choice([int(i) for i in constraints.choices])
+
         min_ = constraints.min_perf if self.perf_mode else constraints.min
         max_ = constraints.max_perf if self.perf_mode else constraints.max
-        min_eval = self.eval_var(min_)
-        if constraints.is_size:
-            min_eval = max(0, min_eval)
-        value = random.randint(min_eval, self.eval_var(max_))
+        if name + '_min' in self.specs:
+            assert(min_ < self.specs[name + '_min'])
```
Comment (Shaac) [inline iorgen/random_input.py]: `min_` is not necessary an integer. You have to use `self.eval_var(min_)`

File: iorgen/random_input.py [hunk 2]
```diff
@@ -40,14 +41,28 @@ def eval_var(self, var: Union[int, str, Variable]) -> int:
...
+            min_ = self.specs[name + '_min']
+        if name + '_max' in self.specs:
+            assert(max_ > self.specs[name + '_max'])
```
Comment (Shaac) [inline iorgen/random_input.py]: Same, use `self.eval_var(max_)`

File: iorgen/random_input.py [hunk 3]
```diff
@@ -40,14 +41,28 @@ def eval_var(self, var: Union[int, str, Variable]) -> int:
...
+            max_ = self.specs[name + '_max']
+
+        if name in self.specs:
+            assert(self.specs[name] >= min_ and self.specs[name] <= max_
```
Comment (Shaac) [inline iorgen/random_input.py]: Same, use `self.eval_var`

File: README.md [hunk 4]
```diff
@@ -48,6 +48,12 @@ Several options are available:
   that it matches the format described in the input YAML.
 - The `--generate_random` also changes the behavior. It will not generate a
   parser, but a valid possible raw input.
+- The `--specify` is to be used with `--generate_random`, it allows the user
+  to specify the value, the min value or the max value of a variable for this
+  output.
+  ```shell
+  $ iorgen -g -s N_max 10 integer 42 -- example.yaml
```
[the remark this answers is not in the document]
Comment (Melyodas) [reply README.md]: `integer 42` set the value of `integer` to 42` whereas `N_max` set the maximum value.

File: iorgen/random_input.py [hunk 5]
```diff
@@ -75,7 +90,7 @@ def generate_line(self, name: str, type_: Type,
             size = self.eval_var(type_.size)
             if type_.encapsulated.main == TypeEnum.INT:
                 return " ".join(
-                    self.generate_integer("", constraints)
+                    self.generate_integer(name, constraints)
```
[the remark this answers is not in the document]
Comment (Melyodas) [reply iorgen/random_input.py]: It is in order to access specifications for the list. I didn't know how to do that without the name.

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (Melyodas) [reply README.md]: Is it better now ?

[the remark this answers is not in the document]
Comment (Shaac) [reply README.md]: yes!

**Pull request merged**